Fix key_test unit test

During cleanup to remove `logurs` from GUAC (as it is not being used) and pass `context` to allow for the logger to be initialized later as needed, the unit test in `key_test.go` failed. 

```
--- FAIL: TestRetrieve (0.22s)
    key_test.go:390: the key provider is being overwritten: mock1
    key_test.go:394: the key provider is being overwritten: mock2
    --- FAIL: TestRetrieve/one_provider (0.00s)
        key_test.go:169: DSSEProcessor.Unpack() = &{SHA256:DoOlWGZKZUoygFf7dJY0aQixmdMf7JNZKlBS6LQmM8A ecdsa 0x140001bd530 ecdsa-sha2-nistp256}, expected &{SHA256:U1OxUJEVw9d+oxhMBqLuqRdj2MsabgK9kb3EgCuTFCM ecdsa 0x140001bd620 ecdsa-sha2-nistp256}
    --- FAIL: TestRetrieve/two_provider (0.00s)
        key_test.go:169: DSSEProcessor.Unpack() = &{SHA256:5L+wqCcJwRO0lulSRokcLbWV6bWgy1l3Ffy1YFfzthY rsa 0x1400010c7e0 rsassa-pss-sha256}, expected &{SHA256:REWE5MQXyLXMaTon1xo2P0DrCRdxA8Kx+Qr2Q4VVtQ4 rsa 0x1400009c240 rsassa-pss-sha256}
```

The issue is that the testing is done via two key providers and where the keys got replaced before the test completes. Fixed via not re-initializing the mocked key providers if they already exist within the `keyProviders` dictionary.

--- a/go.mod
+++ b/go.mod
@@ -8,7 +8,6 @@ require (
 	github.com/in-toto/in-toto-golang v0.3.4-0.20211211042327-af1f9fb822bf
 	github.com/neo4j/neo4j-go-driver/v4 v4.4.4
 	github.com/secure-systems-lab/go-securesystemslib v0.4.0
-	github.com/sirupsen/logrus v1.9.0
 	github.com/spf13/cobra v1.6.1
 	go.uber.org/zap v1.23.0
 	google.golang.org/api v0.103.0
@@ -88,6 +87,7 @@ require (
 	github.com/robfig/cron v1.2.0 // indirect
 	github.com/shurcooL/githubv4 v0.0.0-20201206200315-234843c633fa // indirect
 	github.com/shurcooL/graphql v0.0.0-20200928012149-18c5c3165e3a // indirect
+	github.com/sirupsen/logrus v1.9.0 // indirect
 	github.com/theupdateframework/go-tuf v0.5.2-0.20220930112810-3890c1e7ace4 // indirect
 	github.com/titanous/rocacheck v0.0.0-20171023193734-afe73141d399 // indirect
 	github.com/vbatts/tar-split v0.11.2 // indirect

--- a/internal/testing/ingestor/testdata/testdata.go
+++ b/internal/testing/ingestor/testdata/testdata.go
@@ -16,6 +16,7 @@
 package testdata
 
 import (
+	"context"
 	"encoding/base64"
 	"encoding/json"
 	"reflect"
@@ -438,7 +439,7 @@ func NewMockSigstoreVerifier() *mockSigstoreVerifier {
 	return &mockSigstoreVerifier{}
 }
 
-func (m *mockSigstoreVerifier) Verify(payloadBytes []byte) ([]verifier.Identity, error) {
+func (m *mockSigstoreVerifier) Verify(ctx context.Context, payloadBytes []byte) ([]verifier.Identity, error) {
 
 	keyHash, _ := dsse.SHA256KeyID(ecdsaPubKey)
 	return []verifier.Identity{

--- a/pkg/handler/collector/gcs/gcs.go
+++ b/pkg/handler/collector/gcs/gcs.go
@@ -29,7 +29,7 @@ import (
 	"google.golang.org/api/option"
 
 	"github.com/guacsec/guac/pkg/handler/processor"
-	"github.com/sirupsen/logrus"
+	"github.com/guacsec/guac/pkg/logging"
 )
 
 type gcs struct {
@@ -142,6 +142,7 @@ func (g *gcs) RetrieveArtifacts(ctx context.Context, docChannel chan<- *processo
 }
 
 func (g *gcs) getArtifacts(ctx context.Context, docChannel chan<- *processor.Document) error {
+	logger := logging.FromContext(ctx)
 	it, err := g.reader.getIterator(ctx)
 	if err != nil {
 		return fmt.Errorf("failed to get reader for object for bucket: %s, error: %w", g.bucket, err)
@@ -158,13 +159,13 @@ func (g *gcs) getArtifacts(ctx context.Context, docChannel chan<- *processor.Doc
 		if g.lastDownload.IsZero() {
 			payload, err = g.getObject(ctx, attrs.Name)
 			if err != nil {
-				logrus.Warnf("failed to retrieve object: %s from bucket: %s", attrs.Name, g.bucket)
+				logger.Warnf("failed to retrieve object: %s from bucket: %s", attrs.Name, g.bucket)
 				continue
 			}
 		} else if attrs.Updated.After(g.lastDownload) {
 			payload, err = g.getObject(ctx, attrs.Name)
 			if err != nil {
-				logrus.Warnf("failed to retrieve object: %s from bucket: %s", attrs.Name, g.bucket)
+				logger.Warnf("failed to retrieve object: %s from bucket: %s", attrs.Name, g.bucket)
 				continue
 			}
 		}

--- a/pkg/ingestor/key/inmemory/inmemory.go
+++ b/pkg/ingestor/key/inmemory/inmemory.go
@@ -16,36 +16,40 @@
 package inmemory
 
 import (
+	"context"
+
 	"github.com/guacsec/guac/pkg/ingestor/key"
-	"github.com/sirupsen/logrus"
+	"github.com/guacsec/guac/pkg/logging"
 )
 
 type inmemory struct {
-	collector map[string]key.Key
+	collector map[string]*key.Key
 }
 
 func newInmemoryProvider() *inmemory {
 	return &inmemory{
-		collector: map[string]key.Key{},
+		collector: map[string]*key.Key{},
 	}
 }
 
-func (m *inmemory) RetrieveKey(id string) (*key.Key, error) {
+func (m *inmemory) RetrieveKey(ctx context.Context, id string) (*key.Key, error) {
 	if key, ok := m.collector[id]; ok {
-		return &key, nil
+		return key, nil
 	}
 	return nil, nil
 }
 
-func (m *inmemory) StoreKey(id string, pk *key.Key) error {
-	m.collector[id] = *pk
-	logrus.Warnf("key is being overwritten: %s", id)
+func (m *inmemory) StoreKey(ctx context.Context, id string, pk *key.Key) error {
+	logger := logging.FromContext(ctx)
+	m.collector[id] = pk
+	logger.Warnf("key is being overwritten: %s", id)
 	return nil
 }
 
-func (m *inmemory) DeleteKey(id string) error {
+func (m *inmemory) DeleteKey(ctx context.Context, id string) error {
+	logger := logging.FromContext(ctx)
 	delete(m.collector, id)
-	logrus.Warnf("key is being deleted: %s", id)
+	logger.Warnf("key is being deleted: %s", id)
 	return nil
 }
 

--- a/pkg/ingestor/key/inmemory/inmemory_test.go
+++ b/pkg/ingestor/key/inmemory/inmemory_test.go
@@ -16,20 +16,23 @@
 package inmemory
 
 import (
+	"context"
 	"reflect"
 	"strings"
 	"testing"
 
 	"github.com/guacsec/guac/internal/testing/ingestor/keyutil"
 	"github.com/guacsec/guac/pkg/ingestor/key"
+	"github.com/guacsec/guac/pkg/logging"
 	"github.com/secure-systems-lab/go-securesystemslib/dsse"
 )
 
 func Test_inmemory_RetrieveKey(t *testing.T) {
+	ctx := logging.WithLogger(context.Background())
 	provider, pubKey := setupProvider(t)
-	provider.collector["ecdsa"] = *pubKey[0]
-	provider.collector["rsa"] = *pubKey[1]
-	provider.collector["ed25519"] = *pubKey[2]
+	provider.collector["ecdsa"] = pubKey[0]
+	provider.collector["rsa"] = pubKey[1]
+	provider.collector["ed25519"] = pubKey[2]
 	tests := []struct {
 		name    string
 		id      string
@@ -58,7 +61,7 @@ func Test_inmemory_RetrieveKey(t *testing.T) {
 	}}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			got, err := provider.RetrieveKey(tt.id)
+			got, err := provider.RetrieveKey(ctx, tt.id)
 			if (err != nil) != tt.wantErr {
 				t.Errorf("inmemory.RetrieveKey() error = %v, wantErr %v", err, tt.wantErr)
 				return
@@ -74,6 +77,7 @@ func Test_inmemory_RetrieveKey(t *testing.T) {
 }
 
 func Test_inmemory_StoreKey(t *testing.T) {
+	ctx := logging.WithLogger(context.Background())
 	provider, pubKey := setupProvider(t)
 	type args struct {
 		id string
@@ -111,12 +115,12 @@ func Test_inmemory_StoreKey(t *testing.T) {
 	}}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			err := provider.StoreKey(tt.args.id, tt.args.pk)
+			err := provider.StoreKey(ctx, tt.args.id, tt.args.pk)
 			if (err != nil) != tt.wantErr {
 				t.Errorf("inmemory.StoreKey() error = %v, wantErr %v", err, tt.wantErr)
 			}
 			if err == nil {
-				got, _ := provider.RetrieveKey(tt.args.id)
+				got, _ := provider.RetrieveKey(ctx, tt.args.id)
 				if !reflect.DeepEqual(got, tt.want) {
 					t.Errorf("inmemory.RetrieveKey() = %v, expected %v", got, tt.want)
 				}
@@ -126,10 +130,11 @@ func Test_inmemory_StoreKey(t *testing.T) {
 }
 
 func Test_inmemory_DeleteKey(t *testing.T) {
+	ctx := logging.WithLogger(context.Background())
 	provider, pubKey := setupProvider(t)
-	provider.collector["ecdsa"] = *pubKey[0]
-	provider.collector["rsa"] = *pubKey[1]
-	provider.collector["ed25519"] = *pubKey[2]
+	provider.collector["ecdsa"] = pubKey[0]
+	provider.collector["rsa"] = pubKey[1]
+	provider.collector["ed25519"] = pubKey[2]
 	tests := []struct {
 		name    string
 		id      string
@@ -149,7 +154,7 @@ func Test_inmemory_DeleteKey(t *testing.T) {
 	}}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			err := provider.DeleteKey(tt.id)
+			err := provider.DeleteKey(ctx, tt.id)
 			if (err != nil) != tt.wantErr {
 				t.Errorf("inmemory.DeleteKey() error = %v, wantErr %v", err, tt.wantErr)
 			}

--- a/pkg/ingestor/key/key.go
+++ b/pkg/ingestor/key/key.go
@@ -16,6 +16,7 @@
 package key
 
 import (
+	"context"
 	"crypto"
 	"crypto/ecdsa"
 	"crypto/ed25519"
@@ -26,7 +27,6 @@ import (
 
 	"github.com/secure-systems-lab/go-securesystemslib/dsse"
 	"github.com/sigstore/sigstore/pkg/cryptoutils"
-	"github.com/sirupsen/logrus"
 )
 
 type KeyType string
@@ -47,13 +47,13 @@ type KeyProvider interface {
 	// and it retrieves the wrapped key struct that is associated
 	// Returns nil, nil if no keys are found
 	// Return nil, error if the request to the provider failed
-	RetrieveKey(id string) (*Key, error)
+	RetrieveKey(ctx context.Context, id string) (*Key, error)
 	// StoreKey takes in the ID and the crypto.PublicKey and stores them
 	// for future retrieval. If key id is already present, it replaces
 	// the key with the new one
-	StoreKey(id string, pk *Key) error
+	StoreKey(ctx context.Context, id string, pk *Key) error
 	// DeleteKey takes in the ID and will remove the associated key from the provider
-	DeleteKey(id string) error
+	DeleteKey(ctx context.Context, id string) error
 	// Type returns the key provider type
 	Type() KeyProviderType
 }
@@ -73,20 +73,21 @@ var (
 	keyProviders = map[KeyProviderType]KeyProvider{}
 )
 
-func RegisterKeyProvider(k KeyProvider, providerType KeyProviderType) {
+func RegisterKeyProvider(k KeyProvider, providerType KeyProviderType) error {
 	if _, ok := keyProviders[providerType]; ok {
-		logrus.Warnf("the key provider is being overwritten: %s", providerType)
+		return fmt.Errorf("the key provider is being overwritten: %s", providerType)
 	}
 	keyProviders[providerType] = k
+	return nil
 }
 
 // Find goes through each of the registered key providers and retrieves the wrapped Key
 // TODO: Should this handle if multiple keys are returned
-func Find(id string) (*Key, error) {
+func Find(ctx context.Context, id string) (*Key, error) {
 	var foundKey *Key
 	var err error
 	for i := range keyProviders {
-		foundKey, err = Retrieve(id, i)
+		foundKey, err = Retrieve(ctx, id, i)
 		if err != nil && !strings.Contains(err.Error(), "failed to find key from key provider") {
 			return nil, err
 		}
@@ -101,11 +102,11 @@ func Find(id string) (*Key, error) {
 }
 
 // Retrieve goes to the specified key provider and gets the wrapped Key
-func Retrieve(id string, providerType KeyProviderType) (*Key, error) {
+func Retrieve(ctx context.Context, id string, providerType KeyProviderType) (*Key, error) {
 	var pubKey *Key
 	var err error
 	if provider, ok := keyProviders[providerType]; ok {
-		pubKey, err = provider.RetrieveKey(id)
+		pubKey, err = provider.RetrieveKey(ctx, id)
 		if err != nil {
 			return nil, fmt.Errorf("failed retrieval of key from %s, with error %w", providerType, err)
 		}
@@ -119,7 +120,7 @@ func Retrieve(id string, providerType KeyProviderType) (*Key, error) {
 // Store goes to the specified key provider and stores the wrapped Key.
 // It takes in a PEM-encoded byte slice and converts it to a wrapped Key type
 // returns a nil error when successful
-func Store(id string, pemBytes []byte, providerType KeyProviderType) error {
+func Store(ctx context.Context, id string, pemBytes []byte, providerType KeyProviderType) error {
 	key, err := cryptoutils.UnmarshalPEMToPublicKey(pemBytes)
 	if err != nil {
 		return err
@@ -139,7 +140,7 @@ func Store(id string, pemBytes []byte, providerType KeyProviderType) error {
 		Scheme: KeyScheme,
 	}
 	if provider, ok := keyProviders[providerType]; ok {
-		err := provider.StoreKey(id, foundKey)
+		err := provider.StoreKey(ctx, id, foundKey)
 		if err != nil {
 			return fmt.Errorf("failed storing of key to %s, with error %w", providerType, err)
 		}
@@ -151,9 +152,9 @@ func Store(id string, pemBytes []byte, providerType KeyProviderType) error {
 
 // Delete goes to the specified key provider and deletes the Key
 // returns a nil error when successful
-func Delete(id string, providerType KeyProviderType) error {
+func Delete(ctx context.Context, id string, providerType KeyProviderType) error {
 	if provider, ok := keyProviders[providerType]; ok {
-		err := provider.DeleteKey(id)
+		err := provider.DeleteKey(ctx, id)
 		if err != nil {
 			return fmt.Errorf("failed deleting of key from %s, with error %w", providerType, err)
 		}

--- a/pkg/ingestor/key/key_test.go
+++ b/pkg/ingestor/key/key_test.go
@@ -16,18 +16,21 @@
 package key
 
 import (
+	"context"
 	"crypto"
 	"reflect"
 	"strings"
 	"testing"
 
 	"github.com/guacsec/guac/internal/testing/ingestor/keyutil"
+	"github.com/guacsec/guac/pkg/logging"
 	"github.com/secure-systems-lab/go-securesystemslib/dsse"
 )
 
 func TestFind_OneProvider(t *testing.T) {
+	ctx := logging.WithLogger(context.Background())
 	provider, _, wantKey := setupOneProvider(t)
-	provider.collector["ecdsa"] = *wantKey
+	provider.collector["ecdsa"] = wantKey
 	tests := []struct {
 		name    string
 		id      string
@@ -45,7 +48,7 @@ func TestFind_OneProvider(t *testing.T) {
 	}}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			got, err := Find(tt.id)
+			got, err := Find(ctx, tt.id)
 			if (err != nil) != tt.wantErr {
 				t.Errorf("Find() error = %v, wantErr %v", err, tt.wantErr)
 				return
@@ -61,9 +64,10 @@ func TestFind_OneProvider(t *testing.T) {
 }
 
 func TestFind_MultiProvider(t *testing.T) {
+	ctx := logging.WithLogger(context.Background())
 	provider, _, wantKey := setupTwoProvider(t)
-	provider[0].collector["ecdsa"] = *wantKey[0]
-	provider[1].collector["rsa"] = *wantKey[1]
+	provider[0].collector["ecdsa"] = wantKey[0]
+	provider[1].collector["rsa"] = wantKey[1]
 	tests := []struct {
 		name    string
 		id      string
@@ -86,7 +90,7 @@ func TestFind_MultiProvider(t *testing.T) {
 	}}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			got, err := Find(tt.id)
+			got, err := Find(ctx, tt.id)
 			if (err != nil) != tt.wantErr {
 				t.Errorf("Find() error = %v, wantErr %v", err, tt.wantErr)
 				return
@@ -101,9 +105,10 @@ func TestFind_MultiProvider(t *testing.T) {
 }
 
 func TestRetrieve(t *testing.T) {
+	ctx := logging.WithLogger(context.Background())
 	provider, _, wantKey := setupTwoProvider(t)
-	provider[0].collector["ecdsa"] = *wantKey[0]
-	provider[1].collector["rsa"] = *wantKey[1]
+	provider[0].collector["ecdsa"] = wantKey[0]
+	provider[1].collector["rsa"] = wantKey[1]
 	type args struct {
 		id           string
 		providerType KeyProviderType
@@ -149,7 +154,7 @@ func TestRetrieve(t *testing.T) {
 	}}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			got, err := Retrieve(tt.args.id, tt.args.providerType)
+			got, err := Retrieve(ctx, tt.args.id, tt.args.providerType)
 			if (err != nil) != tt.wantErr {
 				t.Errorf("Retrieve() error = %v, wantErr %v", err, tt.wantErr)
 				return
@@ -169,6 +174,7 @@ func TestRetrieve(t *testing.T) {
 }
 
 func TestStore(t *testing.T) {
+	ctx := logging.WithLogger(context.Background())
 	_, pubBytes, wantKey := setupTwoProvider(t)
 	type args struct {
 		id           string
@@ -211,7 +217,7 @@ func TestStore(t *testing.T) {
 	}}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			err := Store(tt.args.id, tt.args.pk, tt.args.providerType)
+			err := Store(ctx, tt.args.id, tt.args.pk, tt.args.providerType)
 			if (err != nil) != tt.wantErr {
 				t.Errorf("Store() error = %v, wantErr %v", err, tt.wantErr)
 			}
@@ -220,7 +226,7 @@ func TestStore(t *testing.T) {
 					t.Errorf("Retrieve() error = %s, wantErrMessage %s", err, tt.wantErrMessage)
 				}
 			} else {
-				found, _ := Find(tt.args.id)
+				found, _ := Find(ctx, tt.args.id)
 				if !reflect.DeepEqual(found, tt.want) {
 					t.Errorf("DSSEProcessor.Unpack() = %v, expected %v", found, tt.want)
 				}
@@ -230,14 +236,21 @@ func TestStore(t *testing.T) {
 }
 
 func TestDelete(t *testing.T) {
+	ctx := logging.WithLogger(context.Background())
 	provider := newMockProvider()
 	provider2 := newMockProvider()
-	RegisterKeyProvider(provider, "mock1")
-	RegisterKeyProvider(provider2, "mock2")
-	provider.collector["ecdsa"] = Key{
+	err := RegisterKeyProvider(provider, "mock4")
+	if err != nil {
+		t.Log(err)
+	}
+	err = RegisterKeyProvider(provider2, "mock5")
+	if err != nil {
+		t.Log(err)
+	}
+	provider.collector["ecdsa"] = &Key{
 		Type: "ecdsa",
 	}
-	provider2.collector["rsa"] = Key{
+	provider2.collector["rsa"] = &Key{
 		Type: "rsa",
 	}
 	type args struct {
@@ -253,28 +266,28 @@ func TestDelete(t *testing.T) {
 		name: "delete in provider 1",
 		args: args{
 			id:           "ecdsa",
-			providerType: "mock1",
+			providerType: "mock4",
 		},
 		wantErr: false,
 	}, {
-		name: "store in provider 2",
+		name: "delete in provider 2",
 		args: args{
 			id:           "rsa",
-			providerType: "mock2",
+			providerType: "mock5",
 		},
 		wantErr: false,
 	}, {
 		name: "provider not found",
 		args: args{
 			id:           "rsa",
-			providerType: "mock3",
+			providerType: "mock6",
 		},
 		wantErr:        true,
-		wantErrMessage: "key provider not initialized for mock3",
+		wantErrMessage: "key provider not initialized for mock6",
 	}}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			err := Delete(tt.args.id, tt.args.providerType)
+			err := Delete(ctx, tt.args.id, tt.args.providerType)
 			if (err != nil) != tt.wantErr {
 				t.Errorf("Delete() error = %v, wantErr %v", err, tt.wantErr)
 			}
@@ -297,28 +310,28 @@ func TestDelete(t *testing.T) {
 }
 
 type mockKeyProvider struct {
-	collector map[string]Key
+	collector map[string]*Key
 }
 
 func newMockProvider() *mockKeyProvider {
 	return &mockKeyProvider{
-		collector: map[string]Key{},
+		collector: map[string]*Key{},
 	}
 }
 
-func (m *mockKeyProvider) RetrieveKey(id string) (*Key, error) {
+func (m *mockKeyProvider) RetrieveKey(ctx context.Context, id string) (*Key, error) {
 	if key, ok := m.collector[id]; ok {
-		return &key, nil
+		return key, nil
 	}
 	return nil, nil
 }
 
-func (m *mockKeyProvider) StoreKey(id string, pk *Key) error {
-	m.collector[id] = *pk
+func (m *mockKeyProvider) StoreKey(ctx context.Context, id string, pk *Key) error {
+	m.collector[id] = pk
 	return nil
 }
 
-func (m *mockKeyProvider) DeleteKey(id string) error {
+func (m *mockKeyProvider) DeleteKey(ctx context.Context, id string) error {
 	delete(m.collector, id)
 	return nil
 }
@@ -333,8 +346,10 @@ func setupOneProvider(t *testing.T) (*mockKeyProvider, []byte, *Key) {
 		t.Fatalf("failed to get ecdsa key. Error: %v", err)
 	}
 	provider := newMockProvider()
-	RegisterKeyProvider(provider, "mock1")
-
+	err = RegisterKeyProvider(provider, "mock1")
+	if err != nil {
+		t.Log(err)
+	}
 	keyHash, err := dsse.SHA256KeyID(ecdsaPub)
 	if err != nil {
 		t.Fatal("failed to get key hash for test")
@@ -368,10 +383,28 @@ func setupTwoProvider(t *testing.T) ([]*mockKeyProvider, [][]byte, []*Key) {
 		t.Fatalf("failed to get ed25519 key. Error: %v", err)
 	}
 
-	provider := newMockProvider()
-	provider2 := newMockProvider()
-	RegisterKeyProvider(provider, "mock1")
-	RegisterKeyProvider(provider2, "mock2")
+	var provider *mockKeyProvider
+	var provider2 *mockKeyProvider
+
+	if foundProvider, ok := keyProviders["mock1"]; ok {
+		provider = foundProvider.(*mockKeyProvider)
+	} else {
+		provider = newMockProvider()
+		err = RegisterKeyProvider(provider, "mock1")
+		if err != nil {
+			t.Log(err)
+		}
+	}
+
+	if foundProvider, ok := keyProviders["mock2"]; ok {
+		provider2 = foundProvider.(*mockKeyProvider)
+	} else {
+		provider2 = newMockProvider()
+		err = RegisterKeyProvider(provider2, "mock2")
+		if err != nil {
+			t.Log(err)
+		}
+	}
 
 	keyHash, err := dsse.SHA256KeyID(ecdsaPub)
 	if err != nil {

--- a/pkg/ingestor/parser/dsse/parser_dsse.go
+++ b/pkg/ingestor/parser/dsse/parser_dsse.go
@@ -42,15 +42,15 @@ func NewDSSEParser() common.DocumentParser {
 // Parse breaks out the document into the graph components
 func (d *dsseParser) Parse(ctx context.Context, doc *processor.Document) error {
 	d.doc = doc
-	err := d.getIdentity()
+	err := d.getIdentity(ctx)
 	if err != nil {
 		return err
 	}
 	return nil
 }
 
-func (d *dsseParser) getIdentity() error {
-	identities, err := verifier.VerifyIdentity(d.doc)
+func (d *dsseParser) getIdentity(ctx context.Context) error {
+	identities, err := verifier.VerifyIdentity(ctx, d.doc)
 	if err != nil {
 		return err
 	}

--- a/pkg/ingestor/verifier/sigstore_verifier/sigstore_verifier.go
+++ b/pkg/ingestor/verifier/sigstore_verifier/sigstore_verifier.go
@@ -17,6 +17,7 @@ package sigstore_verifier
 
 import (
 	"bytes"
+	"context"
 	"crypto"
 	"encoding/json"
 	"fmt"
@@ -38,14 +39,14 @@ func NewSigstoreVerifier() *sigstoreVerifier {
 
 // Verify validates that the signature is valid for the payload
 // TODO: this currently only supports SHA256 hash function when validating signatures
-func (d *sigstoreVerifier) Verify(payloadBytes []byte) ([]verifier.Identity, error) {
+func (d *sigstoreVerifier) Verify(ctx context.Context, payloadBytes []byte) ([]verifier.Identity, error) {
 	identities := []verifier.Identity{}
 	envelope, err := parseDSSE(payloadBytes)
 	if err != nil {
 		return nil, err
 	}
 	for _, signature := range envelope.Signatures {
-		key, err := key.Find(signature.KeyID)
+		key, err := key.Find(ctx, signature.KeyID)
 		if err != nil {
 			return nil, err
 		}

--- a/pkg/ingestor/verifier/sigstore_verifier/sigstore_verifier_test.go
+++ b/pkg/ingestor/verifier/sigstore_verifier/sigstore_verifier_test.go
@@ -17,6 +17,7 @@ package sigstore_verifier
 
 import (
 	"bytes"
+	"context"
 	"crypto"
 	"crypto/ecdsa"
 	"crypto/rsa"
@@ -32,6 +33,7 @@ import (
 	"github.com/guacsec/guac/pkg/handler/processor"
 	"github.com/guacsec/guac/pkg/ingestor/key"
 	"github.com/guacsec/guac/pkg/ingestor/verifier"
+	"github.com/guacsec/guac/pkg/logging"
 	"github.com/in-toto/in-toto-golang/in_toto"
 	slsa "github.com/in-toto/in-toto-golang/in_toto/slsa_provenance/v0.2"
 	"github.com/sigstore/sigstore/pkg/signature"
@@ -48,19 +50,19 @@ func newMockProvider() *mockKeyProvider {
 	}
 }
 
-func (m *mockKeyProvider) RetrieveKey(id string) (*key.Key, error) {
+func (m *mockKeyProvider) RetrieveKey(ctx context.Context, id string) (*key.Key, error) {
 	if key, ok := m.collector[id]; ok {
 		return &key, nil
 	}
 	return nil, nil
 }
 
-func (m *mockKeyProvider) StoreKey(id string, pk *key.Key) error {
+func (m *mockKeyProvider) StoreKey(ctx context.Context, id string, pk *key.Key) error {
 	m.collector[id] = *pk
 	return nil
 }
 
-func (m *mockKeyProvider) DeleteKey(id string) error {
+func (m *mockKeyProvider) DeleteKey(ctx context.Context, id string) error {
 	delete(m.collector, id)
 	return nil
 }
@@ -70,14 +72,18 @@ func (m *mockKeyProvider) Type() key.KeyProviderType {
 }
 
 func setupOneProvider(t *testing.T) {
+	ctx := logging.WithLogger(context.Background())
 	provider := newMockProvider()
-	key.RegisterKeyProvider(provider, "mock")
+	err := key.RegisterKeyProvider(provider, "mock")
+	if err != nil {
+		t.Log(err)
+	}
 
-	err := key.Store(ecdsaKeyID, []byte(ecdsaPub), "mock")
+	err = key.Store(ctx, ecdsaKeyID, []byte(ecdsaPub), "mock")
 	if err != nil {
 		t.Fatal("failed to store into mock key provider")
 	}
-	err = key.Store(rsaKeyID, []byte(rsapub), "mock")
+	err = key.Store(ctx, rsaKeyID, []byte(rsapub), "mock")
 	if err != nil {
 		t.Fatal("failed to store into mock key provider")
 	}
@@ -94,12 +100,13 @@ func randomData(t *testing.T, n int) []byte {
 }
 
 func TestSigstoreVerifier_Verify(t *testing.T) {
+	ctx := logging.WithLogger(context.Background())
 	setupOneProvider(t)
-	foundECDSAKey, err := key.Find(ecdsaKeyID)
+	foundECDSAKey, err := key.Find(ctx, ecdsaKeyID)
 	if err != nil {
 		t.Fatal("failed to find key in mock key provider")
 	}
-	foundRSAKey, err := key.Find(rsaKeyID)
+	foundRSAKey, err := key.Find(ctx, rsaKeyID)
 	if err != nil {
 		t.Fatal("failed to find key in mock key provider")
 	}
@@ -208,7 +215,7 @@ func TestSigstoreVerifier_Verify(t *testing.T) {
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			sigVerifier := NewSigstoreVerifier()
-			got, err := sigVerifier.Verify(tt.doc.Blob)
+			got, err := sigVerifier.Verify(ctx, tt.doc.Blob)
 			if (err != nil) != tt.wantErr {
 				t.Errorf("SigstoreVerifier.Verify() error = %v, wantErr %v", err, tt.wantErr)
 				return
@@ -226,12 +233,13 @@ func TestSigstoreVerifier_Verify(t *testing.T) {
 }
 
 func TestMultiSignatureSigstoreVerifier_Verify(t *testing.T) {
+	ctx := logging.WithLogger(context.Background())
 	setupOneProvider(t)
-	foundECDSAKey, err := key.Find(ecdsaKeyID)
+	foundECDSAKey, err := key.Find(ctx, ecdsaKeyID)
 	if err != nil {
 		t.Fatal("failed to find key in mock key provider")
 	}
-	foundRSAKey, err := key.Find(rsaKeyID)
+	foundRSAKey, err := key.Find(ctx, rsaKeyID)
 	if err != nil {
 		t.Fatal("failed to find key in mock key provider")
 	}
@@ -325,7 +333,7 @@ func TestMultiSignatureSigstoreVerifier_Verify(t *testing.T) {
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			sigVerifier := NewSigstoreVerifier()
-			got, err := sigVerifier.Verify(tt.doc.Blob)
+			got, err := sigVerifier.Verify(ctx, tt.doc.Blob)
 			if (err != nil) != tt.wantErr {
 				t.Errorf("SigstoreVerifier.Verify() error = %v, wantErr %v", err, tt.wantErr)
 				return

--- a/pkg/ingestor/verifier/verifier.go
+++ b/pkg/ingestor/verifier/verifier.go
@@ -16,6 +16,7 @@
 package verifier
 
 import (
+	"context"
 	"fmt"
 
 	"github.com/guacsec/guac/pkg/handler/processor"
@@ -36,7 +37,7 @@ type Verifier interface {
 	// about the unverified identity. Upstream caller is expected to know
 	// which verifier it needs to based on what type of enveloper or
 	// signature is being verified.
-	Verify(payloadBytes []byte) ([]Identity, error)
+	Verify(ctx context.Context, payloadBytes []byte) ([]Identity, error)
 	// Type returns the verifier type
 	Type() VerifierType
 }
@@ -66,11 +67,11 @@ func RegisterVerifier(k Verifier, providerType VerifierType) error {
 }
 
 // VerifyIdentity goes through the registered providers and verifies the signatures in the payload
-func VerifyIdentity(doc *processor.Document) ([]Identity, error) {
+func VerifyIdentity(ctx context.Context, doc *processor.Document) ([]Identity, error) {
 	switch doc.Type {
 	case processor.DocumentDSSE:
 		if verifier, ok := verifierProviders["sigstore"]; ok {
-			return verifier.Verify(doc.Blob)
+			return verifier.Verify(ctx, doc.Blob)
 		}
 	}
 	return nil, fmt.Errorf("failed verification for document type: %s", doc.Type)

--- a/pkg/ingestor/verifier/verifier_test.go
+++ b/pkg/ingestor/verifier/verifier_test.go
@@ -16,6 +16,7 @@
 package verifier
 
 import (
+	"context"
 	"encoding/base64"
 	"encoding/json"
 	"reflect"
@@ -25,6 +26,7 @@ import (
 	"github.com/guacsec/guac/internal/testing/ingestor/keyutil"
 	"github.com/guacsec/guac/pkg/handler/processor"
 	"github.com/guacsec/guac/pkg/ingestor/key"
+	"github.com/guacsec/guac/pkg/logging"
 	"github.com/secure-systems-lab/go-securesystemslib/dsse"
 )
 
@@ -98,7 +100,7 @@ func newMockSigstoreVerifier() *mockSigstoreVerifier {
 	return &mockSigstoreVerifier{}
 }
 
-func (m *mockSigstoreVerifier) Verify(payloadBytes []byte) ([]Identity, error) {
+func (m *mockSigstoreVerifier) Verify(ctx context.Context, payloadBytes []byte) ([]Identity, error) {
 	keyHash, err := dsse.SHA256KeyID(ecdsaPubKey)
 	if err != nil {
 		return nil, err
@@ -122,6 +124,7 @@ func (m *mockSigstoreVerifier) Type() VerifierType {
 }
 
 func TestVerifyIdentity(t *testing.T) {
+	ctx := logging.WithLogger(context.Background())
 	err := RegisterVerifier(newMockSigstoreVerifier(), "sigstore")
 	if err != nil {
 		t.Errorf("RegisterVerifier() failed with error: %v", err)
@@ -166,7 +169,7 @@ func TestVerifyIdentity(t *testing.T) {
 	}}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			got, err := VerifyIdentity(tt.doc)
+			got, err := VerifyIdentity(ctx, tt.doc)
 			if (err != nil) != tt.wantErr {
 				t.Errorf("VerifyIdentity() error = %v, wantErr %v", err, tt.wantErr)
 				return

